### PR TITLE
Pytest conversion: unit/analysis/stats

### PR DESCRIPTION
**Pull request by ukmo-ccbunney**
### Description
Pytest conversion of files in unit/analysis/stats, comprising the single file:
  - test_pearsonr.py

### CODE REVIEW

File: lib/iris/tests/unit/analysis/stats/test_pearsonr.py [hunk 1]
```diff
@@ -16,15 +12,21 @@
 import iris._lazy_data
 import iris.analysis.stats as stats
 from iris.exceptions import CoordinateNotFoundError
+from iris.tests._shared_utils import (
+    assert_array_almost_equal,
+    assert_array_equal,
+    assert_masked_array_equal,
+    get_data_path,
+    skip_data,
+)
```
Comment (ESadek-MO) [inline lib/iris/tests/unit/analysis/stats/test_pearsonr.py]:
```suggestion
from iris.tests import _shared_utils
```

(Including the function calls)

File: lib/iris/tests/unit/analysis/stats/test_pearsonr.py [hunk 2]
```diff
@@ -139,7 +141,7 @@ def test_broadcast_transpose_cubes_weighted(self, mocked_compute):
 
         mocked_compute.assert_not_called()
         # Should get the same result, but transposed.
-        np.testing.assert_array_almost_equal(r_test.data, r_ref.data.T)
+        assert_array_almost_equal(r_test.data, r_ref.data.T)
 
     def test_weight_error(self):
         with pytest.raises(ValueError):
```
Comment (ESadek-MO) [inline lib/iris/tests/unit/analysis/stats/test_pearsonr.py]: `PT011` Ruff failure here. 

All pytest.raises() need a match as well (excluding some edge cases). So:

`with pytest.raises(ValueError, match=msg)`

Comment (ESadek-MO) [reply lib/iris/tests/unit/analysis/stats/test_pearsonr.py]: (Another failure on line 216, change that too please!)

Comment (ESadek-MO) [reply lib/iris/tests/unit/analysis/stats/test_pearsonr.py]: And 224!

**Review by ESadek-MO — CHANGES_REQUESTED** [review]
A couple changes.

File: lib/iris/tests/unit/analysis/stats/test_pearsonr.py (hunk 2 above)
Comment (ukmo-ccbunney) [reply lib/iris/tests/unit/analysis/stats/test_pearsonr.py]: Oh - funny how my pre-commit didn't catch this. Perhaps it needs updating!

Comment (ESadek-MO) [reply lib/iris/tests/unit/analysis/stats/test_pearsonr.py]: It's not a pre-commit thing. Iris excludes the PTs explicitly, I think! So you need to run No.15 in the checklist especially

Comment (ukmo-ccbunney) [reply lib/iris/tests/unit/analysis/stats/test_pearsonr.py]: Done in 7674ccece

File: lib/iris/tests/unit/analysis/stats/test_pearsonr.py (hunk 1 above)
Comment (ukmo-ccbunney) [reply lib/iris/tests/unit/analysis/stats/test_pearsonr.py]: Done in ded44d4
